dubbo-serialization/dubbo-serialization-msgpack/src/test/java/org/apache/dubbo/common/serialize/msgpack/MsgpackObjectOutputTest.java

```diff
@@ -0,0 +1,161 @@
+    }
+
+    @Test
+    public void testWriteInt() throws IOException {
+        this.msgpackObjectOutput.writeInt(1);
+        this.flushToInput();
+
+        assertThat(msgpackObjectInput.readInt(), is(1));
+    }
+
+    @Test
+    public void testWriteLong() throws IOException {
+        this.msgpackObjectOutput.writeLong(1000L);
+        this.flushToInput();
+
+        assertThat(msgpackObjectInput.readLong(), is(1000L));
+    }
+
+    @Test
+    public void testWriteUTF() throws IOException {
+        this.msgpackObjectOutput.writeUTF("Pace Hasîtî 和平 Мир");
+        this.flushToInput();
+
+        assertThat(msgpackObjectInput.readUTF(), is("Pace Hasîtî 和平 Мир"));
+    }
+
+    @Test
+    public void testWriteUTF2() throws IOException {
+        this.msgpackObjectOutput.writeUTF("a");
+        this.msgpackObjectOutput.writeUTF("b");
+        this.msgpackObjectOutput.writeUTF("c");
+        this.flushToInput();
+        assertThat(msgpackObjectInput.readUTF(), is("a"));
+        assertThat(msgpackObjectInput.readUTF(), is("b"));
+        assertThat(msgpackObjectInput.readUTF(), is("c"));
+    }
+
+
+    @Test
+    public void testWriteThrowable() throws IOException, ClassNotFoundException {
+        BizException throwable = new BizException("error");
+        this.msgpackObjectOutput.writeThrowable(throwable);
+        this.flushToInput();
+        Throwable ex = msgpackObjectInput.readThrowable();
+        assertThat(ex.getMessage(), is("error"));
+        assertThat(ex.getClass(), is(BizException.class));
+    }
+
+    @Test
+    public void testWriteFloat() throws IOException {
+        this.msgpackObjectOutput.writeFloat(1.88f);
+        this.flushToInput();
+
+        assertThat(this.msgpackObjectInput.readFloat(), is(1.88f));
+    }
+
+    @Test
+    public void testWriteDouble() throws IOException {
+        this.msgpackObjectOutput.writeDouble(1.66d);
+        this.flushToInput();
+
+        assertThat(this.msgpackObjectInput.readDouble(), is(1.66d));
+    }
+
+    @Test
+    public void testWriteBytes() throws IOException {
+        this.msgpackObjectOutput.writeBytes("hello".getBytes());
+        this.flushToInput();
+
+        assertThat(this.msgpackObjectInput.readBytes(), is("hello".getBytes()));
+    }
+
+    @Test
+    public void testWriteBytesWithSubLength() throws IOException {
+        this.msgpackObjectOutput.writeBytes("hello".getBytes(), 2, 2);
+        this.flushToInput();
+
+        assertThat(this.msgpackObjectInput.readBytes(), is("ll".getBytes()));
+    }
+
+    @Test
+    public void testWriteByte() throws IOException {
+        this.msgpackObjectOutput.writeByte((byte) 123);
+        this.flushToInput();
+
+        assertThat(this.msgpackObjectInput.readByte(), is((byte) 123));
+    }
+
+    @Test
+    public void testWriteObject() throws IOException, ClassNotFoundException {
+        Image image = new Image("http://dubbo.apache.org/img/dubbo_white.png", "logo", 300, 480, Image.Size.SMALL);
```

Comment:
Please save this file to test resources. Try to reduce network dependence for unit test cases.

Reply:
get